### PR TITLE
server/asset/firo: block parsing for median fees

### PR DESCRIPTION
**Pull request by buck54321**
On testnet (v0.14.12.4), neither `estimatefee` nor `estimatesmartfee` give results, so we fall back to using median fees. Firo doesn't have `getblockstats`, so we have to do `ManualMedianFee` and scan blocks, but we didn't have a `BlockFeeTransactions` function set, so we were using the default `wire.MsgBlock.Deserialize`, which doesn't work for Firo. This made Firo unusable as a backend on testnet.

This doesn't appear to be a problem on mainnet (v0.14.12.1) or simnet (v0.14.12.4). **Note that v0.14.12.4 shouldn't be run on mainnet**

### CODE REVIEW

**Comment by dev-warrior777** [comment]
I have asked @levonpetrosyan93 from 'firoorg' to comment

File: server/asset/firo/firo.go [hunk 1]
```diff
@@ -106,3 +122,211 @@ func NewBackend(configPath string, logger dex.Logger, network dex.Network) (asse
...
+			}
+
+			if _, err := io.ReadFull(r, blk.HashRootMTP[:]); err != nil {
+				return err
+			}
+
+			if blk.HashRootMTP != [16]byte{} {
+				// I have scanned the pre-ProgPOW sections of both testnet
+				// and mainnet and have yet to find non-zero HashRootMTP. I am
+				// speculating that the MTP proof data was purged in an upgrade
```
Comment (dev-warrior777) [inline server/asset/firo/firo.go:294]: This is exactly correct. Later code purged unneeded MTP data.

File: server/asset/firo/firo.go [hunk 2]
```diff
@@ -106,3 +122,211 @@ func NewBackend(configPath string, logger dex.Logger, network dex.Network) (asse
...
+		feeTxs[i] = &btc.BTCFeeTx{MsgTx: msgTx}
+	}
+	return feeTxs, firoBlock.Header.PrevBlock, nil
+}
+
+func deserializeFiroBlock(b []byte, net dex.Network) (*FiroBlock, error) {
+	r := bytes.NewReader(b)
+	blk := NewFiroBlock(net)
+	if err := blk.readBlockHeader(r); err != nil {
+		return nil, fmt.Errorf("readBlocHeader error: %v", err)
```
Comment (JoeGruffins) [inline server/asset/firo/firo.go:184]:
```suggestion
		return nil, fmt.Errorf("readBlockHeader error: %v", err)
```

File: server/asset/firo/firo.go [hunk 3]
```diff
@@ -106,3 +122,211 @@ func NewBackend(configPath string, logger dex.Logger, network dex.Network) (asse
...
+		nType := (tx.Version >> 16) & 0xffff
+		const txTypeNormal = 0
+		if nType != txTypeNormal {
+			// vExtraPayload
+			sz, err := wire.ReadVarInt(r, pver)
+			if err != nil {
+				return nil, fmt.Errorf("ReadVarInt error: %v", err)
+			}
+			if _, err = io.CopyN(io.Discard, r, int64(sz)); err != nil {
+				return nil, fmt.Errorf("error discaring vExtraPayload: %v", err)
```
Comment (JoeGruffins) [inline server/asset/firo/firo.go:213]:
```suggestion
				return nil, fmt.Errorf("error discarding vExtraPayload: %v", err)
```

**Review by JoeGruffins — APPROVED** [review]

**Pull request merged**